server/service/src/processors/transfer/invoice/create_inbound_invoice.rs

```diff
@@ -60,40 +65,80 @@ impl InvoiceTransferProcessor for CreateInboundInvoiceProcessor {
-            outbound_invoice.invoice_row.status,
+            &outbound_invoice.invoice_row.status,
             InvoiceStatus::Shipped | InvoiceStatus::Picked
         ) {
-            return Ok(None);
+            return Ok(InvoiceTransferOutput::WrongOutboundStatus(
+                outbound_invoice.invoice_row.status.to_owned(),
+            ));
         }
+
         // 4.
         if linked_invoice.is_some() {
-            return Ok(None);
+            return Ok(InvoiceTransferOutput::NoLinkedInvoice);
         }
+
         // 5.
         let store = StoreRowRepository::new(connection)
             .find_one_by_id(&record_for_processing.other_party_store_id)?
             .ok_or(RepositoryError::NotFound)?;
-
         if let Some(created_date) = store.created_date {
-            let store_created_datetime = NaiveDateTime::new(
-                created_date - Duration::days(30),
-                NaiveTime::from_hms_opt(0, 0, 0).unwrap_or_default(),
-            );
+            let store_created_datetime =
+                NaiveDateTime::new(created_date - Duration::days(30), Default::default());
             let invoice_created_datetime = outbound_invoice.invoice_row.created_datetime;
             if invoice_created_datetime < store_created_datetime {
-                return Ok(None);
+                return Ok(InvoiceTransferOutput::InvoiceCreatedBeforeStore);
+            }
+        }
+
+        // 6.
+        if outbound_invoice.invoice_row.status == InvoiceStatus::Picked {
+            if let Some(picked_date) = outbound_invoice.invoice_row.picked_datetime {
+                let pref_months = PreventTransfersMonthsBeforeInitialisation {}
+                    .load(connection, None)
+                    .map_err(|e| RepositoryError::DBError {
+                        msg: e.to_string(),
+                        extra: "".to_string(),
+                    })?;
+                if pref_months > 0 {
+                    let sort = Sort {
+                        key: SyncLogSortField::DoneDatetime,
+                        desc: None,
+                    };
+
+                    let latest_log_sorted_by_finished_datetime = SyncLogRepository::new(connection)
+                        .query(Pagination::one(), None, Some(sort))?
+                        .pop();
+
+                    if let Some(log) = latest_log_sorted_by_finished_datetime {
+                        if let Some(initialisation_date) =
+                            log.sync_log_row.integration_finished_datetime
+                        {
+                            if let Some(cutoff_date) = initialisation_date
+                                .checked_sub_months(Months::new(pref_months as u32))
+                            {
+                                if picked_date < cutoff_date {
+                                    return Ok(InvoiceTransferOutput::BeforeInitialisationMonths);
+                                }
+                            }
+                        }
```

Comment:
I think at this point i would make a separate method, just to reduce cognitive load of this processor, still keeping it somewhat 'high' level